.github/workflows/ci.yaml

```diff
@@ -172,4 +172,16 @@ jobs:
         if: ${{ github.event_name != 'pull_request' }}
         run: |
           docker buildx imagetools inspect ghcr.io/${{ github.repository }}:${{ steps.meta.outputs.version }}
-
+  deploy:
+    needs: [ docker ]
+    runs-on: ubuntu-latest
+    if: github.ref == 'refs/heads/main'
```

Comment:
I'm not sure about the semantics here, but I think this will run on every PR (line 7) to main, even unmerged ones. If so this is not the desired behavior.

Reply:
You are correct. For future reference: The github.ref variable condition in this case checks if the current branch is main, so the commit must already be on main. https://docs.github.com/en/actions/learn-github-actions/contexts#determining-when-to-use-contexts